### PR TITLE
Add nil checks on the controller module for underlying app

### PR DESCRIPTION
**Pull request by chatton**
## Description

closes: #2040

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [x] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

**Review by charleenfei — APPROVED** [review]
nice one! 🎉

File: modules/apps/27-interchain-accounts/controller/ibc_middleware.go [hunk 1]
```diff
@@ -101,7 +103,10 @@ func (im IBCMiddleware) OnChanOpenAck(
 	}
 
 	// call underlying app's OnChanOpenAck callback with the counterparty app version.
-	return im.app.OnChanOpenAck(ctx, portID, channelID, counterpartyChannelID, counterpartyVersion)
+	if im.app != nil {
+		return im.app.OnChanOpenAck(ctx, portID, channelID, counterpartyChannelID, counterpartyVersion)
+	}
+	return nil
 }
```
Comment (colin-axner) [inline modules/apps/27-interchain-accounts/controller/ibc_middleware.go:110]:
```suggestion
	if im.app != nil {
		return im.app.OnChanOpenAck(ctx, portID, channelID, counterpartyChannelID, counterpartyVersion)
	}
	
	return nil
}
```
nit: I think it is nice just to space these out

File: modules/apps/27-interchain-accounts/controller/ibc_middleware_test.go [hunk 2]
```diff
@@ -121,6 +122,7 @@ func SetupICAPath(path *ibctesting.Path, owner string) error {
 
 func (suite *InterchainAccountsTestSuite) TestOnChanOpenInit() {
 	var channel *channeltypes.Channel
+	var isNilApp bool
```
Comment (colin-axner) [inline modules/apps/27-interchain-accounts/controller/ibc_middleware_test.go:125]: we can group these vars:
```go
var (


)
```

File: modules/apps/27-interchain-accounts/controller/ibc_middleware_test.go [hunk 3]
```diff
@@ -170,6 +177,8 @@ func (suite *InterchainAccountsTestSuite) TestOnChanOpenInit() {
 		suite.Run(tc.name, func() {
 			suite.SetupTest() // reset
 
+			isNilApp = false
```
Comment (colin-axner) [inline modules/apps/27-interchain-accounts/controller/ibc_middleware_test.go:180]:
```suggestion
			suite.SetupTest() // reset
			isNilApp = false
```

File: modules/apps/27-interchain-accounts/controller/ibc_middleware_test.go [hunk 4]
```diff
@@ -623,6 +663,10 @@ func (suite *InterchainAccountsTestSuite) TestOnTimeoutPacket() {
 			cbs, ok := suite.chainA.App.GetIBCKeeper().Router.GetRoute(module)
 			suite.Require().True(ok)
 
+			if isNilApp {
```
Comment (colin-axner) [inline modules/apps/27-interchain-accounts/controller/ibc_middleware_test.go:666]: Could you explicitly set to false? I believe if a test case is appended, it'll still be set to true

File: modules/apps/27-interchain-accounts/controller/ibc_middleware_test.go [hunk 5]
```diff
@@ -327,6 +346,10 @@ func (suite *InterchainAccountsTestSuite) TestOnChanOpenAck() {
 
 			err = cbs.OnChanOpenAck(suite.chainA.GetContext(), path.EndpointA.ChannelConfig.PortID, path.EndpointA.ChannelID, path.EndpointB.ChannelID, path.EndpointB.ChannelConfig.Version)
 
+			if isNilApp {
```
Comment (colin-axner) [inline modules/apps/27-interchain-accounts/controller/ibc_middleware_test.go:349]: explicitly set to false in test case loop

**Review by colin-axner — COMMENTED** [review]
LGTM, left only comments on code formatting

File: modules/apps/27-interchain-accounts/controller/ibc_middleware_test.go (hunk 4 above)
Comment (chatton) [reply modules/apps/27-interchain-accounts/controller/ibc_middleware_test.go:666]: nice catch, missed this on a few of the test functions

**Review by colin-axner — APPROVED** [review]
Excellent! Thank you!

**Comment by colin-axner** [comment]
Could you add a changelog entry indicating nil apps are allowed in ics27 controller middleware?

**Review by seantking — APPROVED** [review]

**Pull request merged**